game/data/weapons.py

```diff
@@ -267,7 +268,9 @@ def for_aircraft(cls, aircraft: AircraftType, number: int) -> Pylon:
                 pylon_number, weapon = value
                 if pylon_number != number:
                     continue
-                allowed.add(Weapon.with_clsid(weapon["clsid"]))
+                allowed_weapon = Weapon.with_clsid(weapon["clsid"])
+                if allowed_weapon is not None:
```

Comment:
Do we actually expect this? If we can't find a weapon for a pylon that DCS says is there, that sounds like a bug.

Reply:
That's not necessarily DCS telling us it's there but rather pydcs telling us it doesn't know about the weapon and thus won't be able to load the weapon. I can think of two different:

- An old loadout that was scanned by pydcs with an invalid CLSID, invalid as in outdated/obsolete and is thus no longer present in the weapons map
- A new loadout that was scanned after a newly released version of DCS while not having updated pydcs yet, which was recently the case with TALDs

Also, may I remind you that it was you who suggested to ignore payloads that can't be loaded in #2320, so your comment is really confusing me...